src/vs/platform/keybinding/common/keybindingResolver.ts

```diff
@@ -162,12 +157,15 @@ export class KeybindingResolver {
 				continue;
 			}
 
-			const conflictHasMultipleChords = (conflict.chords.length > 1);
-			const itemHasMultipleChords = (item.chords.length > 1);
-
-			// TODO@chords
-			if (conflictHasMultipleChords && itemHasMultipleChords && conflict.chords[1] !== item.chords[1]) {
-				// The conflict only shares the first chord with this command
+			let diverges = false;
+			for (let i = 1; i < conflict.chords.length && i < item.chords.length; i++) {
```

Comment:
hm, I'm not sure whether difference in length between `item.chords` and `conflict.chords` would be a "divergence" here

If `conflict` is `cmd+k cmd+k` and `item` is `cmd+k cmd+k cmd+k`, `conflict` would not be removed from lookup, but also unclear to me whether a user would want it to be

Reply:
I think the logic here makes sense, although it's a little subtle when the keybindings have differing lengths. (I renamed the variable to `chordsFullyOverlap`, which hopefully makes it a bit more clear?)

Essentially, if the keybinding overlaps completely (as in the shorter keybinding is fully contained within the longer one), they logically conflict, since when the chord sequence is pressed, it triggers the shorter keybinding, then exits the chord mode. In this case, the shorter keybinding completely shadows the longer one. (And I think it makes sense to flag this in the UI + treat it in the same way as other conflicts, with the later defined binding taking precedence).

I will also leave a comment to that effect here? (Feel free to edit the comment if you think it's confusing)!

Reply:
E.g. for your example: `cmd+k cmd+k cmd+k` and `cmd+k cmd+k`, we'd find that `shorterChordIsPrefix = true`, and so we won't fall out of the loop iteration via continue, perform the `whenIsEntirelyIncluded` check, and treat the keybinding as a conflict.

(At least that's my understanding of the code as written, please correct me if I'm wrong about how this interacts with the rest of the system 😅)

Reply:
Makes sense, thanks!